File: qooxdoo-contrib/RpcPython/trunk/services/start-server.py
#!/usr/bin/python
# -*- coding: ascii -*-
#############################################################################
#
#   RpcPython
#
#   http://qooxdoo.org/contrib/project#rpcpython
#
#   Copyright:
#      
#
#   License:
#     LGPL: http:/f/www.gnu.org/licenses/lgpl.html
#     EPL: http://www.eclipse.org/org/documents/epl-v10.php
#     See the LICENSE file in the project's top-level directory for details.
#
#   Authors:
#     * Viktor Ferenczi (python@cx.hu)
#     * Christian Boulanger (cboulanger)
#
#############################################################################

'''==========================================================================

This is the main server that listens for requests, imports service classes
and calls the service methods. It receives a service name in dot-separated path 
format and expect to find the class containing the service in a file. 
If the service name is "foo.bar.baz", the class is named "Baz" in the 
"foo.bar.baz" module, located in "foo/bar/baz.py" somewhere on the python
class path. The class file is dynamically loaded when the request is received. 
The classes and methods are protected. The service class is loaded only if
the containing module contains the "isRpcService" property set to True and
if the method contains the  "public" decorator.

=========================================================================='''

import sys
import os
import qxjsonrpc
from server.jsonrpc import JsonRpcServer

# Add "class" subdirectory to pythonpath
service_class_path = os.path.dirname(os.path.abspath(__file__)) + "/class"
sys.path.append( service_class_path )

def main():
    '''Run json-rpc server.'''
    print 'Running: qxjsonrpc %s'%qxjsonrpc.__version__.number 
    print
    print 'Debugging output is enabled in the test server, watch stdout.'
    print
    print 'Ctrl-C aborts this server.'
    print
    print 'Server log output follows:'
    print
    
    #@todo get host and port from command line arguments
    host = '127.0.0.1'
    port=8000
    debug=True;
    
    #start server
    srv=JsonRpcServer(host,port,debug)
    srv.serve_forever()

#============================================================================

if __name__=='__main__':
    main()

#============================================================================

File: qooxdoo-contrib/RpcNode/trunk/demo/default/source/class/rpcnode/demo/start-server.js
/* ************************************************************************

   JSON-RPC 1.0 and 2.0 implementation running on node.js
   
   Copyright:
     2010 The autors
     
   License:
     LGPL: http://www.gnu.org/licenses/lgpl.html
     EPL: http://www.eclipse.org/org/documents/epl-v10.php
     See the LICENSE file in the project's top-level directory for details.

   Authors:
     * Martin Wittemann (original implementation)
     * Christian Boulanger (port to qooxdoo-contrib & to qooxdoo-rpc )
   
   Dependencies:
     * node-promise (http://github.com/kriszyp/node-promise)

************************************************************************ */

/*
 * imports
 */
libpath  = "../../../../../../";
require( libpath + 'source/class/rpcnode/Server' );
require( "./service/NodeTest" );
require( "./service/RpcTest" );

/*
 * create new server on port 8888
 */
var server = new rpcnode.Server(8888);

/*
 * add anonymous services (strict 1.0/2.0):
 * only the method names will be used in figuring out the service 
 * method
 */
//console.log(nodejs.sys.inspect(rpcnode));
server.addService( new rpcnode.demo.service.NodeTest() );
server.addService( new rpcnode.demo.service.RpcTest() );

/*
 * add named services (non-standard json-rpc, maybe version 2.1)
 */
server.addService( new rpcnode.demo.service.NodeTest, "nodetest" );
server.addService( new rpcnode.demo.service.RpcTest, "qooxdoo.test" );

/*
 * start the server
 */
server.start();
  